### ISSUE TITLE
join(...).onKey(TableField) doesn't work with aliased tables

### ISSUE THREAD

**Issue by arlampin** [jOOQ/jOOQ#7626]
### Expected behavior and actual behavior:
I expected 
`.onKey(someAlias.SOME_FIELD)` 
to work similar to 
`.onKey(SOME_GENERATED_TABLE.SOME_FIELD)`
but the first one throws `org.jooq.exception.DataAccessException: No matching Key found`.

Plain `.onKey()` works though, as do the `onKey(ForeignKey)` and manually constructed `.on(Condition)`.

### Steps to reproduce the problem (include example code if possible):
```java
Project projectAlias = PROJECT.as("p");
context.select()
    .from(projectAlias)
    .join(ACCOUNT).onKey(projectAlias.ACCOUNT_ID)
    .fetch();
```

Tables are something like:
```sql
create table account (
  id serial primary key
  ...
);
create table project (
  id serial primary key,
  account_id integer not null references account,
  ...
);
```
Stacktrace:
```
Exception in thread "main" org.jooq.exception.DataAccessException: No matching Key found between tables ["p"] and ["public"."account"]
        at org.jooq.impl.JoinTable.onKeyException(JoinTable.java:668)
        at org.jooq.impl.JoinTable.onKey(JoinTable.java:606)
        at org.jooq.impl.JoinTable.onKey(JoinTable.java:132)
        at org.jooq.impl.SelectQueryImpl.addJoinOnKey(SelectQueryImpl.java:2487)
        at org.jooq.impl.SelectImpl.onKey(SelectImpl.java:2102)
        at org.jooq.impl.SelectImpl.onKey(SelectImpl.java:139)
```

### Versions:

- jOOQ: 3.10.7 / 3.11.2
- Java:
- Database (include vendor):
- OS:
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Related: #5209

**Comment by lukaseder** [comment]
#5050 needs to be fixed to fix this

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.14.0. Backport scheduled for 3.13.3 (#10324)
